Documentation/General/emergentArchitectureDesign/emergentArchitectureDesign.tex

```diff
@@ -236,12 +236,12 @@ \section{Scalability}
 The goal is clear: Find a way for the system to scale further.
 We went over several iterations of the system.
 Once we realised the issue with plain Nearest Neighbours, we went for an ANN\footnote{Approximate nearest neighbour} algorithm to be able to use the nearest neighbour algorithm.
-This initially was planned to be implemented using a KD-Tree \cite{KD-Tree}.
+This initially was planned to be implemented using a \gls{KD-tree}.
 However, this would cause memory issues due to a O(N) space complexity of the algorithm.
```

Comment:
It's very offtopic for this Pr, because this is already merged, but shouldn't we depict the fact that:
* The structure has to reside in memory, which is a cons for a large dataset on low system resources / budgets.
* Maintaining the structure with realtime querying and updating is intensive.